src/main/java/dinkplugin/message/DiscordMessageHandler.java

```diff
@@ -162,6 +170,49 @@ private void sendMessage(HttpUrl url, MultipartBody.Builder requestBody, int att
         });
     }
 
+    /**
+     * Captures the next frame and applies the specified rescaling
+     * while abiding by {@link Embed#MAX_IMAGE_SIZE}.
+     *
+     * @param drawManager  {@link DrawManager}
+     * @param scalePercent {@link DinkPluginConfig#screenshotScale()} divided by 100.0
+     * @return future of the image byte array by the image format name
+     * @apiNote scalePercent should be in (0, 1]
+     * @implNote the image format is either "png" (lossless) or "jpeg" (lossy), both of which can be used in MIME type
+     */
+    private static CompletableFuture<Map.Entry<String, byte[]>> captureScreenshot(DrawManager drawManager, double scalePercent) {
+        CompletableFuture<Image> future = new CompletableFuture<>();
+        drawManager.requestNextFrameListener(future::complete);
+        return future.thenApply(ImageUtil::bufferedImageFromImage)
+            .thenApply(input -> Utils.rescale(input, scalePercent))
+            .thenApply(image -> {
+                try {
+                    String format = "png"; // lossless
+                    return Pair.of(format, Utils.convertImageToByteArray(image, format));
+                } catch (IOException e) {
+                    throw new CompletionException("Could not convert image to byte array", e);
+                }
+            })
+            .thenApply(pair -> {
+                byte[] bytes = pair.getValue();
+                int n = bytes.length;
+                if (n <= Embed.MAX_IMAGE_SIZE)
+                    return pair; // already compliant; no further rescale necessary
+
+                // calculate scale factor to comply with MAX_IMAGE_SIZE
+                double factor = Math.sqrt(1.0 * Embed.MAX_IMAGE_SIZE / n);
+
+                // bytes => original image => rescaled image => updated bytes
+                try (InputStream is = new ByteArrayInputStream(bytes)) {
+                    String format = "jpeg"; // lossy
+                    BufferedImage rescaled = Utils.rescale(ImageIO.read(is), factor);
+                    return Pair.of(format, Utils.convertImageToByteArray(rescaled, format));
+                } catch (Exception e) {
+                    throw new CompletionException("Failed to automatically resize image below Discord size limit", e);
```

Comment:
```suggestion
                    throw new CompletionException("Failed to resize image below Discord size limit", e);
```

nitpick: the word `automatically` here doesn't provide anything useful